### PR TITLE
Added tabIndex parameter to dialog buttons

### PR DESCRIPTION
**Pull request by TheOtherDude**
### This project uses [semver](semver.org), please check the scope of this pr:
 - [ ] #patch# - backwards-compatible bug fix
 - [x] #minor# - adding functionality in a backwards-compatible manner
 - [ ] #major# - incompatible API change

# CHANGELOG
* Added tabIndex parameter to dialog buttons
* Gave cancel button a default tabIndex of 1 so that it comes after confirm button in tab order

### CODE REVIEW

**Comment by ember-cli-visual-acceptance** [comment]
# Visual Acceptance tests:

### No changes

File: addon/templates/components/frost-modal-dialog.hbs [hunk 1]
```diff
@@ -82,6 +83,7 @@
       priority=button.priority
       size='medium'
       onClick=button.onClick
+      tabIndex=(ifUndefined button.tabIndex 1 button.tabIndex)
```
Comment (rox163) [inline addon/templates/components/frost-modal-dialog.hbs]: it seems this would give all custom buttons a tabIndex of 1 if that prop is undefined?

Comment (TheOtherDude) [reply addon/templates/components/frost-modal-dialog.hbs]: Oops, good catch!

File: addon/templates/components/frost-modal-dialog.hbs [hunk 2]
```diff
@@ -69,6 +69,7 @@
     isVisible=params.cancel.isVisible
     priority='tertiary'
     size='medium'
+    tabIndex=(ifUndefined params.cancel.tabIndex 1 params.cancel.tabIndex)
```
Comment (sglanzer-fire) [inline addon/templates/components/frost-modal-dialog.hbs]: Can we just use the standard if where undefined would evaluate to falsey?

File: addon/helpers/if-undefined.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import Ember from 'ember';
+
+export function ifUndefined([condition, whenUndef, whenNotUndef, ...rest]) {
+  return condition === undefined ? whenUndef : whenNotUndef
+}
+
+export default Ember.Helper.helper(ifUndefined);
```
Comment (sglanzer-fire) [inline addon/helpers/if-undefined.js]: If we decide this helper is necessary (see below) - I would still like it to come from `core`

**Review by sglanzer-fire — CHANGES_REQUESTED** [review]
Let's confirm if the new helper is necessary

File: addon/templates/components/frost-modal-dialog.hbs (hunk 2 above)
Comment (TheOtherDude) [reply addon/templates/components/frost-modal-dialog.hbs]: Unfortunately, that would mean that a defined tabIndex of zero would be ignored as if it were undefined.

File: addon/helpers/if-undefined.js (hunk 3 above)
Comment (TheOtherDude) [reply addon/helpers/if-undefined.js]: Do I need to do anything besides add the helper to `ember-frost-core`?

File: addon/templates/components/frost-modal-dialog.hbs (hunk 2 above)
Comment (sglanzer-fire) [reply addon/templates/components/frost-modal-dialog.hbs]: Chuck an `(or` in as well?

Comment (sglanzer-fire) [reply addon/templates/components/frost-modal-dialog.hbs]: `(if (or params.cancel.tabIndex (eq params.cancel.tabIndex 0)) params.cancel.tabIndex 1)` I think...

Comment (sglanzer-fire) [reply addon/templates/components/frost-modal-dialog.hbs]: Sorry, I just want to avoid a helper if possible

**Comment by TheOtherDude** [comment]
Removed `ifUndefined` helper. Thanks for the tip @sglanzer

**Comment by sglanzer-fire** [comment]
Glad it worked!  Thanks for making the change 👍

**Comment by sglanzer-fire** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-modal/pull-request/60/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-modal)

**Pull request merged**